### ISSUE TITLE
version 0.3.1 doesn't compile

### ISSUE THREAD

**Issue by InsidiousMind** [tomusdrw/rust-web3#153]
I wasn't having this issue until i did a cargo-update on a crate that uses rust-web3. However, I've tested this in a vagrant VM with Ubuntu 16.04 and am not able to compile web3. I am not entirely sure why the  Travis builds have passed if it fails on my machine/vagrant VM. I think the issue is just some outdated/multiple dependencies with ethereum_types/ethabi/error-chain so i submitted a pull request which fixes these issues.

I get this output when trying to compile: 

```
   Compiling web3 v0.3.1 (file:///home/vagrant/rust-web3)
warning: lint unused_doc_comment has been renamed to unused_doc_comments
  --> src/contract/error.rs:8:1
   |
8  | / error_chain! {
9  | |   links {
10 | |     Abi(ethabi::Error, ethabi::ErrorKind);
11 | |     Api(::Error, ::ErrorKind);
...  |
19 | |   }
20 | | }
   | |_^
   |
   = note: #[warn(renamed_and_removed_lints)] on by default
   = note: this error originates in a macro outside of the current crate (in Nightly builds, run with -Z external-macro-backtrace for more info)

warning: lint unused_doc_comment has been renamed to unused_doc_comments
  --> src/contract/error.rs:8:1
   |
8  | / error_chain! {
9  | |   links {
10 | |     Abi(ethabi::Error, ethabi::ErrorKind);
11 | |     Api(::Error, ::ErrorKind);
...  |
19 | |   }
20 | | }
   | |_^
   |
   = note: this error originates in a macro outside of the current crate (in Nightly builds, run with -Z external-macro-backtrace for more info)

warning: lint unused_doc_comment has been renamed to unused_doc_comments
  --> src/contract/error.rs:8:1
   |
8  | / error_chain! {
9  | |   links {
10 | |     Abi(ethabi::Error, ethabi::ErrorKind);
11 | |     Api(::Error, ::ErrorKind);
...  |
19 | |   }
20 | | }
   | |_^
   |
   = note: this error originates in a macro outside of the current crate (in Nightly builds, run with -Z external-macro-backtrace for more info)

warning: lint unused_doc_comment has been renamed to unused_doc_comments
  --> src/contract/error.rs:8:1
   |
8  | / error_chain! {
9  | |   links {
10 | |     Abi(ethabi::Error, ethabi::ErrorKind);
11 | |     Api(::Error, ::ErrorKind);
...  |
19 | |   }
20 | | }
   | |_^
   |
   = note: this error originates in a macro outside of the current crate (in Nightly builds, run with -Z external-macro-backtrace for more info)

warning: lint unused_doc_comment has been renamed to unused_doc_comments
  --> src/contract/error.rs:26:5
   |
26 | /     error_chain! {
27 | |       links {
28 | |         Api(::Error, ::ErrorKind);
29 | |       }
...  |
36 | |       }
37 | |     }
   | |_____^
   |
   = note: this error originates in a macro outside of the current crate (in Nightly builds, run with -Z external-macro-backtrace for more info)

warning: lint unused_doc_comment has been renamed to unused_doc_comments
  --> src/contract/error.rs:26:5
   |
26 | /     error_chain! {
27 | |       links {
28 | |         Api(::Error, ::ErrorKind);
29 | |       }
...  |
36 | |       }
37 | |     }
   | |_____^
   |
   = note: this error originates in a macro outside of the current crate (in Nightly builds, run with -Z external-macro-backtrace for more info)

warning: lint unused_doc_comment has been renamed to unused_doc_comments
  --> src/contract/error.rs:26:5
   |
26 | /     error_chain! {
27 | |       links {
28 | |         Api(::Error, ::ErrorKind);
29 | |       }
...  |
36 | |       }
37 | |     }
   | |_____^
   |
   = note: this error originates in a macro outside of the current crate (in Nightly builds, run with -Z external-macro-backtrace for more info)

warning: lint unused_doc_comment has been renamed to unused_doc_comments
  --> src/contract/error.rs:26:5
   |
26 | /     error_chain! {
27 | |       links {
28 | |         Api(::Error, ::ErrorKind);
29 | |       }
...  |
36 | |       }
37 | |     }
   | |_____^
   |
   = note: this error originates in a macro outside of the current crate (in Nightly builds, run with -Z external-macro-backtrace for more info)

warning: lint unused_doc_comment has been renamed to unused_doc_comments
  --> src/error.rs:10:1
   |
10 | / error_chain! {
11 | |   foreign_links {
12 | |     Io(io::Error);
13 | |   }
...  |
40 | |   }
41 | | }
   | |_^
   |
   = note: this error originates in a macro outside of the current crate (in Nightly builds, run with -Z external-macro-backtrace for more info)

warning: lint unused_doc_comment has been renamed to unused_doc_comments
  --> src/error.rs:10:1
   |
10 | / error_chain! {
11 | |   foreign_links {
12 | |     Io(io::Error);
13 | |   }
...  |
40 | |   }
41 | | }
   | |_^
   |
   = note: this error originates in a macro outside of the current crate (in Nightly builds, run with -Z external-macro-backtrace for more info)

warning: lint unused_doc_comment has been renamed to unused_doc_comments
  --> src/error.rs:10:1
   |
10 | / error_chain! {
11 | |   foreign_links {
12 | |     Io(io::Error);
13 | |   }
...  |
40 | |   }
41 | | }
   | |_^
   |
   = note: this error originates in a macro outside of the current crate (in Nightly builds, run with -Z external-macro-backtrace for more info)

warning: lint unused_doc_comment has been renamed to unused_doc_comments
  --> src/error.rs:10:1
   |
10 | / error_chain! {
11 | |   foreign_links {
12 | |     Io(io::Error);
13 | |   }
...  |
40 | |   }
41 | | }
   | |_^
   |
   = note: this error originates in a macro outside of the current crate (in Nightly builds, run with -Z external-macro-backtrace for more info)

error[E0308]: mismatched types
   --> src/contract/tokens.rs:181:40
    |
181 |             Token::Address(data) => Ok(data),
    |                                        ^^^^ expected struct `ethereum_types::H160`, found struct `ethereum_types::hash::H160`
    |
    = note: expected type `ethereum_types::H160`
               found type `ethereum_types::hash::H160`
note: Perhaps two different versions of crate `ethereum_types` are being used?
   --> src/contract/tokens.rs:181:40
    |
181 |             Token::Address(data) => Ok(data),
    |                                        ^^^^

error[E0308]: mismatched types
   --> src/contract/tokens.rs:187:24
    |
187 |         Token::Address(self)
    |                        ^^^^ expected struct `ethereum_types::hash::H160`, found struct `ethereum_types::H160`
    |
    = note: expected type `ethereum_types::hash::H160`
               found type `ethereum_types::H160`
note: Perhaps two different versions of crate `ethereum_types` are being used?
   --> src/contract/tokens.rs:187:24
    |
187 |         Token::Address(self)
    |                        ^^^^

error[E0277]: the trait bound `ethereum_types::U256: std::convert::From<ethereum_types::uint::U256>` is not satisfied
   --> src/contract/tokens.rs:196:59
    |
196 |           Token::Int(data) | Token::Uint(data) => Ok(data.into()),
    |                                                           ^^^^ the trait `std::convert::From<ethereum_types::uint::U256>` is not implemented for `ethereum_types::U256`
...
208 | uint_tokenizable!(U256, "U256");
    | -------------------------------- in this macro invocation
    |
    = help: the following implementations were found:
              <ethereum_types::U256 as std::convert::From<&'a ethereum_types::U512>>
              <ethereum_types::U256 as std::convert::From<usize>>
              <ethereum_types::U256 as std::convert::From<u16>>
              <ethereum_types::U256 as std::convert::From<[u8; 32]>>
            and 15 others
    = note: required because of the requirements on the impl of `std::convert::Into<ethereum_types::U256>` for `ethereum_types::uint::U256`

error[E0277]: the trait bound `ethereum_types::U128: std::convert::From<ethereum_types::uint::U256>` is not satisfied
   --> src/contract/tokens.rs:196:59
    |
196 |           Token::Int(data) | Token::Uint(data) => Ok(data.into()),
    |                                                           ^^^^ the trait `std::convert::From<ethereum_types::uint::U256>` is not implemented for `ethereum_types::U128`
...
209 | uint_tokenizable!(U128, "U128");
    | -------------------------------- in this macro invocation
    |
    = help: the following implementations were found:
              <ethereum_types::U128 as std::convert::From<&'a [u8]>>
              <ethereum_types::U128 as std::convert::From<u8>>
              <ethereum_types::U128 as std::convert::From<u16>>
              <ethereum_types::U128 as std::convert::From<u32>>
            and 14 others
    = note: required because of the requirements on the impl of `std::convert::Into<ethereum_types::U128>` for `ethereum_types::uint::U256`

error[E0277]: the trait bound `ethereum_types::uint::U256: std::convert::From<ethereum_types::U256>` is not satisfied
   --> src/contract/tokens.rs:202:26
    |
202 |         Token::Uint(self.into())
    |                          ^^^^ the trait `std::convert::From<ethereum_types::U256>` is not implemented for `ethereum_types::uint::U256`
...
208 | uint_tokenizable!(U256, "U256");
    | -------------------------------- in this macro invocation
    |
    = help: the following implementations were found:
              <ethereum_types::uint::U256 as std::convert::From<&'a [u8; 32]>>
              <ethereum_types::uint::U256 as std::convert::From<&'a ethereum_types::uint::U512>>
              <ethereum_types::uint::U256 as std::convert::From<ethereum_types::hash::H256>>
              <ethereum_types::uint::U256 as std::convert::From<&'a [u8]>>
            and 16 others
    = note: required because of the requirements on the impl of `std::convert::Into<ethereum_types::uint::U256>` for `ethereum_types::U256`

error[E0277]: the trait bound `ethereum_types::uint::U256: std::convert::From<ethereum_types::U128>` is not satisfied
   --> src/contract/tokens.rs:202:26
    |
202 |         Token::Uint(self.into())
    |                          ^^^^ the trait `std::convert::From<ethereum_types::U128>` is not implemented for `ethereum_types::uint::U256`
...
209 | uint_tokenizable!(U128, "U128");
    | -------------------------------- in this macro invocation
    |
    = help: the following implementations were found:
              <ethereum_types::uint::U256 as std::convert::From<&'a [u8; 32]>>
              <ethereum_types::uint::U256 as std::convert::From<&'a ethereum_types::uint::U512>>
              <ethereum_types::uint::U256 as std::convert::From<ethereum_types::hash::H256>>
              <ethereum_types::uint::U256 as std::convert::From<&'a [u8]>>
            and 16 others
    = note: required because of the requirements on the impl of `std::convert::Into<ethereum_types::uint::U256>` for `ethereum_types::U128`

error[E0308]: mismatched types
  --> src/contract/error.rs:8:1
   |
8  | / error_chain! {
9  | |   links {
10 | |     Abi(ethabi::Error, ethabi::ErrorKind);
11 | |     Api(::Error, ::ErrorKind);
...  |
19 | |   }
20 | | }
   | |_^ expected struct `error_chain::State`, found a different struct `error_chain::State`
   |
   = note: expected type `error_chain::State` (struct `error_chain::State`)
              found type `error_chain::State` (struct `error_chain::State`)
note: Perhaps two different versions of crate `error_chain` are being used?
  --> src/contract/error.rs:8:1
   |
8  | / error_chain! {
9  | |   links {
10 | |     Abi(ethabi::Error, ethabi::ErrorKind);
11 | |     Api(::Error, ::ErrorKind);
...  |
19 | |   }
20 | | }
   | |_^
   = note: this error originates in a macro outside of the current crate (in Nightly builds, run with -Z external-macro-backtrace for more info)

error[E0308]: mismatched types
  --> src/contract/error.rs:8:1
   |
8  | / error_chain! {
9  | |   links {
10 | |     Abi(ethabi::Error, ethabi::ErrorKind);
11 | |     Api(::Error, ::ErrorKind);
...  |
19 | |   }
20 | | }
   | |_^ expected enum `std::option::Option`, found struct `error_chain::backtrace::imp::InternalBacktrace`
   |
   = note: expected type `std::option::Option<std::sync::Arc<error_chain::Backtrace>>`
              found type `error_chain::backtrace::imp::InternalBacktrace`
   = note: this error originates in a macro outside of the current crate (in Nightly builds, run with -Z external-macro-backtrace for more info)

error: aborting due to 8 previous errors

Some errors occurred: E0277, E0308.
For more information about an error, try `rustc --explain E0277`.
error: Could not compile `web3`.

To learn more, run the command again with --verbose.
```

**Comment by TinyCalf** [comment]
it seems a higher version of ethabi leads to this problem. ethabi version 5.1.2 uses error-chain version 0.12. I change dependencies to below and compiles web3 correctly:
```
[dependencies]
web3 = "=0.3.1"
error-chain = "=0.11"
ethabi = "=5.1.1"
```

**Comment by InsidiousMind** [comment]
@TinyCalf Yep, it's fixed in #154 :)

**Comment by D4nte** [comment]
@tomusdrw Hi! Could this be published on crates.io please? Thanks for the fix.
